core/src/scene/dataLayer.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "scene/sceneLayer.h"
+#include <string>
+
+namespace Tangram {
+
+// DataLayer represents a top-level layer in the stylesheet, distinct from
+// SceneLayer by its association with a collection within a DataSource
+class DataLayer {
```

Comment:
Hmm.. I am not sure I get this.

How in case of inheritance "DataLayer" will container other "DataLayer"s?

DataLayer being a specialized SceneLayer will still have vector of SceneLayers (subLayers). 
To me inheritance makes more sense.